modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts

```diff
@@ -24,15 +24,18 @@ import * as InsertList from './InsertList';
 
 const wrappedElements = [ 'pre' ];
 
-const shouldPasteContentOnly = (fragment: AstNode, parentNode: Element) => {
-  const firstNode = fragment.firstChild;
-
-  const isAFlattenableTag = Arr.contains(wrappedElements, firstNode.name);
-  const isPastingInTheSameTag = firstNode.name === parentNode.tagName.toLowerCase();
-  const lastNode = fragment.lastChild.attr('data-mce-type') === 'bookmark' ? fragment.lastChild.prev : fragment.lastChild;
-  const isCopingOnlyOneTag = firstNode === lastNode;
-
-  return isCopingOnlyOneTag && isAFlattenableTag && isPastingInTheSameTag;
+const shouldPasteContentOnly = (dom: DOMUtils, fragment: AstNode, parentNode: Element, editor: Editor): boolean => {
+  const firstNode = fragment.firstChild as AstNode;
```

Comment:
We also need to consider if the fragment itself is marked as a CEF element here I believe. So we probably need to add a check such as `firstNode.attr('contenteditable') !== 'false' so that it won't unwrap a CEF element, though I'd double check that's not handled elsewhere already. A good way to test that would be to add a test for it.

- [ ] Add a test for pasting in a CEF pre block